### ISSUE TITLE
ANR upon Opening the Application

### ISSUE THREAD

**Issue by ahmedk92** [realm/realm-java#1242]
Upon just opening the application I get ANR. Here's the relevant (I think) trace in traces.txt.

```
"pool-1-thread-1" prio=5 tid=11 WAIT
  | group="main" sCount=1 dsCount=0 obj=0xb3d68ac8 self=0xb786fb98
  | sysTid=845 nice=0 sched=0/0 cgrp=apps handle=-1215778832
  | state=S schedstat=( 0 10000000 3 ) utm=0 stm=0 core=0
  at java.lang.Object.wait(Native Method)
  - waiting on <0xb3d5ffb0> (a java.lang.ref.ReferenceQueue)
  at java.lang.Object.wait(Object.java:401)
  at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:102)
  at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:73)
  at io.realm.internal.FinalizerRunnable.run(FinalizerRunnable.java:45)
  at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
  at java.util.concurrent.FutureTask.run(FutureTask.java:237)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
  at java.lang.Thread.run(Thread.java:841)
```

**Comment by emanuelez** [comment]
This is not related to your ANR. Realm spawns a thread that takes care of cleaning up native resources. This is the thread you are showing here and it is normal that it sits waiting for the garbage collector to populate the reference queue. You should probably see what's happening in the UI thread instead.

I'm closing this issue because this is expected behavior and not related to your ANR. Please open a new issue if you can associate the ANR in to Realm in the UI thread.

**Comment by ahmedk92** [comment]
Well, it's my fault. It was just a realm migration exception, and I was improperly handling my exceptions.
